### DISCUSSION TITLE
the compiler is out of heap space

### DISCUSSION THREAD

**Discussion by 10100010010001001010** [axmolengine/axmol#1060] [Q&A]
- axmol version: lastest, i think
- devices test on: PC
- developing environments:
   - Visual Studio 2022   
   - cmake version: 3.25

![not_enough_heap-memory_for_compiler](https://user-images.githubusercontent.com/120378517/219046859-63ceb8c4-8a2a-4eac-a2f1-216df5622d24.png)
 
Errors text: "the compiler is out of heap space". This happens when starting a standard project, even when removing extensions. I ran it on a weak computer at work, while the original engine runs the project without problems

**Comment by DelinWorks** [comment]
what's the computer's specs?

**Reply by 10100010010001001010** [reply]
I forgot to write the most important detail that the problem is 4 GB of RAM. But as I have already written the usual cocos 4.0 works fine

**Reply by DelinWorks** [reply]
how much page swap do you have? if it's set to 0mb, then it will crash on this very small ram

**Reply by 10100010010001001010** [reply]
I can only answer tomorrow because I am not at work right now. But still, why doesn't this prevent the cocos project from running, what exactly needs more performance?

**Reply by DelinWorks** [reply]
you mean compiling, right? if not, then what's the exact issue you're talking about?

**Reply by 10100010010001001010** [reply]
I mean run project (ctrl+f5), but anyway, the project compiles to run, doesn't it?

My work computer has a 2.70 GHz dual-core pentium processor with 3.47 GB of RAM

**Reply by DelinWorks** [reply]
that's a very slow machine. Debugging and making projects in that machine would be quite difficult and frustrating. Can you please check if automatic paging is enabled in the OS? Have you tried @halx99 solution down below?

**Reply by halx99** [reply]
Yes, I guess you use a very older PC, at this time developers usually have 64GB RAM PC for work.

**Reply by 10100010010001001010** [reply]
"automatic paging is enabled in the OS?" -> yes
"Have you tried @halx99 solution down below?" -> no, i saw this answer today, tomorrow i will chek if it helps

**Reply by DelinWorks** [reply]
imo you need at least 8gb (I can compile axmol projects with my 8gb laptop just fine) if you can upgrade your work pc from 4gb to 8gb at the very least you could compile and run projects.

P.S. it won't work if your system is 32bit

**Comment by halx99** [comment]
Try pass `-Thost=x64` to cmake

**Reply by 10100010010001001010** [reply]
i have 32bit system, i think its not help.
